### PR TITLE
Regular

### PR DESCRIPTION
**Pull request by JacquesCarette**
Regular Category, Epi and Mono morphisms.  Needed for Pretopos.

### CODE REVIEW

**Comment by TOTBWF** [comment]
Should we maybe pull the `Regular` morphisms into records? Seems like it might be a bit easier to work with long term.

**Comment by JacquesCarette** [comment]
Good point. [And exactly why I'm now switching to doing PRs myself, to get this kind of feedback.]

File: src/Categories/Morphism/Regular.agda [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+open import Categories.Diagram.Equalizer 𝒞
+open import Categories.Diagram.Coequalizer 𝒞
+
+open Category 𝒞
+
+private
+  variable
+    A B : Obj
+
+RegularMono : (f : A ⇒ B) → Set (o ⊔ ℓ ⊔ e)
```
Comment (HuStmpHrrr) [inline src/Categories/Morphism/Regular.agda:30]: would it be better if you define them as records?

Comment (JacquesCarette) [reply src/Categories/Morphism/Regular.agda:30]: Yes, that was @TOTBWF comment too. As soon as I get a minute...

**Pull request merged**